### ISSUE TITLE
git.latest - permissions issue when using non-salt user and SaltStack fileserver URL identity file

### ISSUE THREAD

**Issue by b1naryth1ef** [saltstack/salt#42550]
### Description of Issue/Question

Using the git.latest state with a non-salt user (e.g. not root in our case) and a SaltStack fileserver URL as the identity file results in the state failing due to permissions issues, for example;

```yaml
version_repo:
  git.latest:
    - name: git@github.com:b1naryth1ef/my-dope-repo.git
    - identity: salt://github/id_rsa
    - target: /home/andrei/my_dope_repo
    - force_fetch: True
    - force_reset: True
    - force_checkout: True
    - user: andrei
```

Would result in the state failing;

```
----------
          ID: version_repo
    Function: git.latest
        Name: git@github.com:b1naryth1ef/my-dope-repo.git
      Result: False
     Comment: Failed to check remote refs: Unable to authenticate using identity file:
              
              Warning: Identity file /var/cache/salt/minion/files/base/github/id_rsa not accessible: Permission denied.
              Permission denied (publickey).
              fatal: Could not read from remote repository.
              
              Please make sure you have the correct access rights
              and the repository exists.
     Started: 20:14:35.820407
    Duration: 160.322 ms
     Changes:   

```

As best I can tell, this is due to the fact that the salt file cache has very limited permissions scoped to the salt minion runtime user. Despite the fact that within the git module we ensure that the user we're running git as can access the temporary script we use for cloning, there are no checks or conditions that ensure the user can access the keyfile.

I imagine this is a somewhat standard setup that should be supported by the git module, however I'm not exactly sure how to accomplish this considering putting the keyfile somewhere that the user calling git can see may be seen as a security risk. I'd be happy to submit a PR if someone can toss me in the right direction.

For now, the workaround I have is to clone as the salt user (root) and then have a watch which chowns the cloned directory.

### Versions Report

```
Salt Version:
           Salt: 2017.7.0
 
Dependency Versions:
           cffi: Not Installed
       cherrypy: Not Installed
       dateutil: 1.5
      docker-py: Not Installed
          gitdb: 0.5.4
      gitpython: 0.3.2 RC1
          ioflo: Not Installed
         Jinja2: 2.7.2
        libgit2: Not Installed
        libnacl: Not Installed
       M2Crypto: Not Installed
           Mako: 0.9.1
   msgpack-pure: Not Installed
 msgpack-python: 0.4.6
   mysql-python: 1.2.3
      pycparser: Not Installed
       pycrypto: 2.6.1
   pycryptodome: Not Installed
         pygit2: Not Installed
         Python: 2.7.6 (default, Oct 26 2016, 20:30:19)
   python-gnupg: Not Installed
         PyYAML: 3.10
          PyZMQ: 14.0.1
           RAET: Not Installed
          smmap: 0.8.2
        timelib: Not Installed
        Tornado: 4.2.1
            ZMQ: 4.0.5
 
System Versions:
           dist: Ubuntu 14.04 trusty
         locale: UTF-8
        machine: x86_64
        release: 4.4.0-87-generic
         system: Linux
        version: Ubuntu 14.04 trusty
```

**Comment by garethgreenaway** [comment]
Thanks for the report.  A similar issue was filed against 2016.11 and fixed in #41945.  Taking a look to see why it's still causing problems.

**Comment by garethgreenaway** [comment]
Looks like the fix didn't get merged into the 2017.7.0 release but will definitely be in 2017.7.1.

**Comment by davidlawson** [comment]
Still having the same issue in 2017.7.1.

**Comment by garethgreenaway** [comment]
@davidlawson Unfortunately it looks like the change isn't in 2017.7.1 but I see it in the 2017.7 branch so it will be in 2017.7.2.

**Comment by garethgreenaway** [comment]
This fix is now in later version of 2017.7 and future versions including Oxygen, so I'm going to close out the issue.  Feel free to reopen if the issue persists.  Thanks!

**Comment by taranlu-houzz** [comment]
I'm still having this issue on macOS.

**Versions Report**
Salt Version:
           Salt: 3000.1

Dependency Versions:
           cffi: 1.12.2
       cherrypy: unknown
       dateutil: 2.8.0
      docker-py: Not Installed
          gitdb: 2.0.6
      gitpython: 2.1.15
         Jinja2: 2.10.1
        libgit2: 1.0.0
       M2Crypto: 0.35.2
           Mako: 1.0.7
   msgpack-pure: Not Installed
 msgpack-python: 0.5.6
   mysql-python: Not Installed
      pycparser: 2.19
       pycrypto: 3.8.1
   pycryptodome: Not Installed
         pygit2: 1.2.0
         Python: 3.7.7 (default, Mar 10 2020, 15:43:27)
   python-gnupg: 0.4.4
         PyYAML: 5.1.2
          PyZMQ: 18.0.1
          smmap: 3.0.2
        timelib: 0.2.4
        Tornado: 4.5.3
            ZMQ: 4.3.1

System Versions:
           dist:
         locale: UTF-8
        machine: x86_64
        release: 17.7.0
         system: Darwin
        version: 10.13.6 x86_64
